central/sensor/service/pipeline/virtualmachineindex/pipeline.go

```diff
@@ -92,18 +106,35 @@ func (p *pipelineImpl) Run(ctx context.Context, _ string, msg *central.MsgFromSe
 	}
 
 	// Enrich VM with vulnerabilities
-	err := p.enricher.EnrichVirtualMachineWithVulnerabilities(vm, indexV4)
+	err = p.enricher.EnrichVirtualMachineWithVulnerabilities(vm, indexV4)
 	if err != nil {
 		return errors.Wrapf(err, "failed to enrich VM %s with vulnerabilities", index.GetId())
 	}
 
 	// Store enriched VM
-	if err := p.vmDatastore.UpdateVirtualMachineScan(ctx, vm.GetId(), vm.GetScan()); err != nil {
-		return errors.Wrapf(err, "failed to upsert VM %s to datastore", index.GetId())
+	err = p.vmDatastore.UpdateVirtualMachineScan(ctx, vm.GetId(), vm.GetScan())
```

Comment:
I thought about the changes to this file and I am not super comfortable that the handling of synthetic load is so intertwined with the production code.

If we want to keep this load generation merged, I think we should trigger the changes to the data store where they originally originate - i.e., in sensor.